src/libs/mps_comm/mockup/base_station.cpp

```diff
@@ -32,7 +34,8 @@ MockupBaseStation::get_base(llsf_msgs::BaseColor color)
 	callback_busy_(true);
 	std::lock_guard<std::mutex> lg(queue_mutex_);
 	queue_.push(std::make_tuple([this] { callback_busy_(false); },
-	                            std::chrono::system_clock::now() + std::chrono::seconds(1)));
+	                            std::chrono::system_clock::now()
+	                              + std::chrono::seconds((int)std::ceil(1.f / exec_speed_))));
```

Comment:
The way you round the durations always results in longer durations, e.g., 4 seconds with a speedup of 3 would execute for 2 seconds. Wouldn't it make more sense to use the min of 1 and the computed duration? Something like this:
```suggestion
	                              + std::min(std::chrono::milliseconds(1000),
	                                         std::chrono::milliseconds((int)(1000.1 / exec_speed_)))));
```

In this example, it doesn't really make sense, but with longer durations, it does.

Reply:
True, makes sense to increase the precisision. done in 89b0bc8b6a051e4d0e26218b830cf4c811498ede